conformance/utils/suite/suite.go

```diff
@@ -124,6 +124,10 @@ func (suite *ConformanceTestSuite) Setup(t *testing.T) {
 	t.Logf("Test Setup: Applying base manifests")
 	suite.Applier.MustApplyWithCleanup(t, suite.Client, suite.BaseManifests, suite.GatewayClassName, suite.Cleanup)
 
+	t.Logf("Test Setup: Applying programmatic resources")
+	secret := kubernetes.MustCreateSelfSignedCertSecret(t, "gateway-conformance-web-backend", "certificate", []string{"*"})
```

Comment:
The `*` hostname might stick out, but the changes made in this PR only cover ReferencePolicy. We may want tests in the future that cover the request-response when using a certificate; however, that is outside the scope of this PR.